src/main/java/com/ibanity/apis/client/products/ponto_connect/services/TokenService.java

```diff
@@ -0,0 +1,20 @@
+package com.ibanity.apis.client.products.ponto_connect.services;
+
+import com.ibanity.apis.client.products.ponto_connect.models.Token;
+
+import java.util.Map;
+
+public interface TokenService {
+
+    void revoke(String token, String clientSecret);
+
+    void revoke(Map<String, String> additionalHeaders, String token, String clientSecret);
+
+    Token create(String authorizationCode, String codeVerifier, String redirectUri, String clientSecret);
```

Comment:
I don't where code_verifier is used . Any code_challenge? code_challenge_method? Maybe in a lib?

Reply:
The code_challenge and code_challenge_method are used during the oauth2 authroization flow.
The oauth2 authroization flow is not handled by the ibannity library.
The ibanity library is only used to interact with Ibanity enpoints.
In the case of oauth2, we exposes /token and /revoke token.
Those endpoints make use of code_verifier, authorization_code, client_id and client_secret.
The client_id is a configuration parameter, the other parameters are passed as method parameters.